File: src/compiler/core/symlink.ts
import { Artifact } from "./artifact.ts";

export let symlink = (target: string): Symlink => {
	return new Symlink(target);
};

export class Symlink {
	target: string;

	constructor(target: string) {
		this.target = target;
	}

	async serialize(): Promise<syscall.Symlink> {
		return {
			target: this.target,
		};
	}

	static async deserialize(symlink: syscall.Symlink): Promise<Symlink> {
		return new Symlink(symlink.target);
	}
}

File: src/artifact/symlink.rs
use camino::Utf8PathBuf;

#[derive(
	Clone,
	Debug,
	PartialEq,
	Eq,
	buffalo::Deserialize,
	buffalo::Serialize,
	serde::Deserialize,
	serde::Serialize,
)]
pub struct Symlink {
	#[buffalo(id = 0)]
	pub target: Utf8PathBuf,
}
